### DISCUSSION TITLE
Swift Package that depends on Swift Package with Resources causes SwiftUI Preview to crash

### DISCUSSION THREAD

**Discussion by matt-dewitt** [pointfreeco/isowords#104] [Q&A] [answered]
Hello, been loving the Composable architecture and learning how to use it and work it into my own apps, and I've really been enjoying your modularization approach to splitting separate features into their own Swift packages but I seem to have hit a snag when I try to implement it myself.

I just created a new project to try and work through the issue and right now it's setup with two features: an AuthenticationFeature and an AppFeature. In the AuthenticationFeature I have a custom image to be displayed, which works fine when I build the AuthenticationFeature and try to preview it, but then when I try to pull in the AuthenticationFeature into my AppFeature, SwiftUI previews continue to crash with this error:

`Application Specific Information:
AuthenticationFeature/resource_bundle_accessor.swift:27: Fatal error: unable to find bundle named Inventory_AuthenticationFeature`

I noticed in isowords you have a couple of features with images, but I can't seem to figure out how you got around this issue. Any help would be greatly appreciated. Thanks!

Here is how I currently have my project structured:

<img width="317" alt="Screen Shot 2021-05-03 at 8 21 52 AM" src="https://user-images.githubusercontent.com/66968193/116888477-c4f3b380-abe8-11eb-93e4-c65bd64d7320.png">

```
let package = Package(
    name: "Inventory",
    platforms: [.iOS(.v14)],
    products: [
        .library(name: "AppFeature", targets: ["AppFeature"]),
        .library(name: "AuthenticationFeature", targets: ["AuthenticationFeature"]),
    ],
    dependencies: [
        .package(url: "https://github.com/pointfreeco/swift-composable-architecture.git", .exact("0.18.0"))
    ],
    targets: [
        .target(
            name: "AppFeature",
            dependencies: [
                "AuthenticationFeature",
                .product(name: "ComposableArchitecture", package: "swift-composable-architecture"),
            ]
        ),
        .target(
            name: "AuthenticationFeature",
            dependencies: [
                .product(name: "ComposableArchitecture", package: "swift-composable-architecture"),
            ],
            resources: [.process("Resources/")]
        ),
    ]
)
```

**Comment by andreyz** [comment, marked as answer]
Hi @matt-dewitt , this is a known limitation of default `Bundle.module` accessor auto-generated by Xcode. Have a look at a [workaround here](https://developer.apple.com/forums/thread/664295?answerId=656407022#656407022).

**Reply by matt-dewitt** [reply]
I have tried this fix but it doesn't seem to work for me; perhaps I'm naming my bundles wrong?

**Reply by matt-dewitt** [reply]
Ahh, I think I finally got it! Is there a good way to modularize this bundle extension if I'm going to need it in several packages? Or do i basically have to put this in every feature that has resources?

**Reply by andreyz** [reply]
I haven't found any other way, but to include a new bundle accessor into every package that has resources.

**Reply by matt-dewitt** [reply]
Bleh, that's no fun, but at least it works for now! Hopefully Apple will fix soon...
